src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -332,24 +354,74 @@ public static FormulaValue TextToBoolean(IRContext irContext, StringValue[] args
                 return new BooleanValue(irContext, false);
             }
 
-            return CommonErrors.InvalidBooleanFormatError(irContext);
+            throw new Exception("Invalid Boolean Format.");
```

Comment:
>throw new Exception("Invalid Boolean Format."); [](http://example.com/codeflow?start=12&length=47)

we should avoid exceptions for control flow like this. 
Instead, use a `bool Try*(out value)` pattern and return false.

Reply:
yes, will update.